solr/solrcloud/README.md

```diff
@@ -0,0 +1,178 @@
+
+## Create a SolrCloud cluster
+
+> **Warning**
+> Don't use the options listed in the quickstart as 1) they reference an ingress controller we didn't use and 2) they don't set up authentication, which we need.
+
+For now, use `helm` to install the cluster. set  `$SOLRCLOUD` to whatever you want to call the overall thing -- e.g. `catalog`, `feddocs`, `htsolr`, whatever. this assumes you'll call the solr cloud instance the same thing as the namespace you install it into.
+
+```bash
+export IMAGE_REPO=ghcr.io/hathitrust/catalog-solr
+export SOLRCLOUD=htsolr
+helm install $SOLRCLOUD apache-solr/solr --version 0.5.1 \
+  --namespace $SOLRCLOUD \
+  --create-namespace \
+  --set image.repository=$IMAGE_REPO \
+  --set image.tag=solrcloud-8.11.1 \
+  --set solrOptions.javaMemory="-Xms300m -Xmx300m" \
+  --set solrOptions.security.authenticationType="Basic"
+```
+
+See https://artifacthub.io/packages/helm/apache-solr/solr for other options you might need (e.g. `solroptions.javaopts`, `solroptions.loglevel`, etc)
+
+You can watch the status of the cluster starting up:
+
+```bash
+kubectl -n $SOLRCLOUD get -w pod
+```
+
+Eventually (it may take a while) all solrs should be online. If all the solrs end up in a `CrashLoopBackOff` state, you can try:
+
+```bash
+kubectl -n $SOLRCLOUD rollout restart statefulset $SOLRCLOUD-solrcloud
+```
+
+(You may need to do this a couple times; the issue seems to be related to timeouts with readiness checks with so many things starting up in parallel.)
+
+Once all the solrs are up and ready, you can connect to solr:
+
+```bash
+kubectl -n $SOLRCLOUD port-forward service/$SOLRCLOUD-solrcloud-common 8983:80
+```
+
+This will ask for a username and password, which SolrCloud has automatically configured for us. We can get the password with:
+
+```bash
+SOLR_PASS=$(kubectl -n $SOLRCLOUD get secret htsolr-solrcloud-security-bootstrap -o jsonpath='{.data.admin}' | base64 -d)
+echo $SOLR_PASS
+```
+
+You can now go to http://localhost:8983 and log in with username `admin` and password from above. If you click on "Cloud" you should see three Solr nodes; if you click on "ZK Status" under cloud you should see status "green" with three ZooKeeper nodes.
+
+## Upload configuration to Solr/ZooKeeper
+
+Now that Solr is up and running, we need to get the configuration for our Solr core into Solr. We can do that using the [ConfigSet API](https://solr.apache.org/guide/8_11/configsets-api.html)
+
+To date, we have only used static Solr configuration via a `solrconfig.xml` typically managed alongside the indexing configuration, rather than using the [Schema API](https://solr.apache.org/guide/8_11/schema-api.html) to dynamically define fields.
+
+### Prepare configuration
+
+Zip up the complete core configuration, including `schema.xml` and anything it references. There should not be any subdirectories in the zip; i.e. the files should not be in the `conf` directory.
```

Comment:
The umich catalog (https://github.com/mlibrary/catalog_solr_index/blob/main/biblio/conf/solrconfig.xml) config doesn't have the extra lib directives as in https://github.com/hathitrust/hathitrust_catalog_indexer/blob/main/solr/catalog/conf/solrconfig.xml, so when @niquerio tried using that config as-is with zookeeper & solr cloud it didn't work because it couldn't find some of the libraries in jars in the contrib directories -- maybe those are by default on the jar path in non-solrcloud mode? Either way I'll make a note about it here.

@billdueber On main here, https://github.com/hathitrust/hathitrust_catalog_indexer/commit/112a4c807638b3e1650d48d2587ea05e1c170e75 makes solrconfig.xml like the umich catalog one -- that commit will break using the config in solrcloud if merged into this branch as-is. That's probably not a big issue for now but still worth noting, especially if/when we rebase this branch.